jck/diff/diff.sh

```diff
@@ -1,104 +1,147 @@
+getVersionValue()
+{	
+	VERSION_VALUE1=$(find "${WORKDIR}/${VERSION1}" -maxdepth 1 -mindepth 1 -type d -name "JCK-runtime*" | awk -F'-' '{print $NF}') 
+	VERSION_VALUE2=$(find "${WORKDIR}/${VERSION2}" -maxdepth 1 -mindepth 1 -type d -name "JCK-runtime*" | awk -F'-' '{print $NF}') 	
+}
+
+compareGroups()
 {
 	dirName=$1
-	cd $WORKDIR/$VERSION1/JCK-$dirName-$VERSION_VALUE1/tests
-	echo "Listing test directories under $dirName at: `pwd`" 
-	find . -maxdepth 2 -mindepth 2 -type d > $WORKDIR/logs/$dirName-$VERSION_VALUE1.txt
-	
-	cd $WORKDIR/$VERSION2/JCK-$dirName-$VERSION_VALUE2/tests
-	echo "Listing test directories under $dirName at: `pwd`" 
-	find . -maxdepth 2 -mindepth 2 -type d > $WORKDIR/logs/$dirName-$VERSION_VALUE2.txt
-
-	if cmp -s $WORKDIR/logs/$dirName-$VERSION_VALUE1.txt $WORKDIR/logs/$dirName-$VERSION_VALUE2.txt; then 
-		echo "SAME : '$dirName' folder identical in both given versions: $VERSION1 & $VERSION2"
-	else 
-		diff $WORKDIR/logs/$dirName-$VERSION_VALUE1.txt $WORKDIR/logs/$dirName-$VERSION_VALUE2.txt > $WORKDIR/logs/$dirName-diff.txt 
-		echo "DIFFERENT : '$dirName' folder content are different in two given versions: $VERSION1 & $VERSION2"
+	jdkVersion1TestDir="${WORKDIR}/${VERSION1}/JCK-${dirName}-${VERSION_VALUE1}/tests"
+	jdkVersion2TestDir="${WORKDIR}/${VERSION2}/JCK-${dirName}-${VERSION_VALUE2}/tests"
+
+	rc=0
+	find "${jdkVersion1TestDir}" -maxdepth 2 -mindepth 2 -type d | sed "s|${jdkVersion1TestDir}||" | sort > "$WORKDIR/logs/$dirName-groups-${VERSION_VALUE1}.txt"
+	find "${jdkVersion2TestDir}" -maxdepth 2 -mindepth 2 -type d | sed "s|${jdkVersion2TestDir}||" | sort > "$WORKDIR/logs/$dirName-groups-${VERSION_VALUE2}.txt"
+
+	if [[ "${dirName}" == "runtime" ]]; then
+		find "${jdkVersion1TestDir}/vm/verifier" -maxdepth 1 -mindepth 1 -type d | sed "s|${jdkVersion1TestDir}||" | sort >> "$WORKDIR/logs/$dirName-groups-${VERSION_VALUE1}.txt"
+		find "${jdkVersion2TestDir}/vm/verifier" -maxdepth 1 -mindepth 1 -type d | sed "s|${jdkVersion2TestDir}||" | sort >> "$WORKDIR/logs/$dirName-groups-${VERSION_VALUE2}.txt"
+	fi
+
+	# Compare if playlist test groups are different
+	diff "$WORKDIR/logs/$dirName-groups-$VERSION_VALUE1.txt" "$WORKDIR/logs/$dirName-groups-$VERSION_VALUE2.txt" > "$WORKDIR/logs/$dirName-groups-diff.txt" || rc=$?
+	if [ $rc -eq 0 ]; then
+		echo "SAME GROUPS : Groups under '$dirName' are identical in given versions: $VERSION1 & $VERSION2"
+	else
+		echo "DIFFERENT GROUPS: Groups under '$dirName' are different in two given versions: $VERSION1 & $VERSION2"
 		echo "Please manually investigate the following differences in the two given repositories:"
-		cat $WORKDIR/logs/$dirName-diff.txt
-		return 1
-	fi 
+		cat "$WORKDIR/logs/$dirName-groups-diff.txt"
+	fi
+	return $rc
+}
+
+compareNewTestsInSubGroups() {
+	subGroups=("vm.verifier.instructions" "lang.CLSS" "lang.EXPR" "lang.CONV" "lang.INTF" "lang.LMBD" "lang.DASG" "lang.NAME" "lang.TYPE" "lang.ANNOT" "lang.STMT")
+	cat > "$WORKDIR/logs/NewTestsInSubGroups.txt"
```

Comment:
This line doesn't look right? Do you mean to echo "some header" > "$WORKDIR/logs/NewTestsInSubGroups.txt"

Reply:
No, it's basically to create the file NewTestsInSubGroups.txt. As I tested I found sometimes `diff "${WORKDIR}/${VERSION1}/JCK-${component}-${VERSION_VALUE1}/tests/${subdir}" "${WORKDIR}/${VERSION2}/JCK-${component}-${VERSION_VALUE2}/tests/${subdir}" >> "$WORKDIR/logs/DiffsInSubGroups.txt"` ">>" won't create the file as expected if the file doesn't exist.

Reply:
So yeah that makes sense, but it's just the syntax seem odd, if I try say:
```
cat > file.txt
```
it hangs!!

Reply:
Oops! All I'm thinking is echo, but don't know how it come out with `cat`. Update with touch. Thanks @andrew-m-leonard !